src/widgets/table/table.rs

```diff
@@ -561,21 +596,42 @@ impl Table<'_> {
 
     fn render_header(&self, area: Rect, buf: &mut Buffer, column_widths: &[(u16, u16)]) {
         if let Some(ref header) = self.header {
+            let area = Rect {
+                x: area.x,
+                y: area.y + header.top_margin,
+                width: area.width,
+                height: area.height - header.top_margin,
+            };
```

Comment:
Do you think it might be clearer to instead add some extra unused space in the layout for margins above?
e.g.
```rust
.constraints([
  Constraint::Length(header.top_margin),
  Constraint::Length(header_height)
  Constraint::Length(header_bottom_margin),
  ...
```
(and then ignore those rows and just grab layout[1], [3], and [5])

Doing this could collapse the header / footer methods to a single method:
```rust
fn render_row(row: row, ...)
```
and could be called with
```rust
self.header.map(|row| render_row(row, header_area, ...);
self.footer.map(|row| render_row(row, footer_area, ...);
```

Reply:
Agreed.

Reply:
Sounds good. Let me try first.